management/lifecycle.py

```diff
@@ -205,8 +208,9 @@ def __init__(self, pr, ghapi, repo=None, priority=1, computetag='openff'):
     def _gather_datasets(self):
         files = self.pr.get_files()
         datasets = list(filter(
-            lambda x: glob.fnmatch.fnmatch(os.path.basename(x), DATASET_GLOB),
-            map(lambda x: x.filename, files)))
+            lambda x: any(glob.fnmatch.fnmatch(os.path.basename(x), match) for match in [DATASET_GLOB, SCAFFOLD_GLOB]),
+            map(lambda x: x.filename, files)
+        ))
```

Comment:
I think this might need to be a list concatenation to avoid messing up the iteration depth (but am not positive, hard to follow everything going on in this command) ... Also this is turning into a pretty gnarly list comprehension that'll be hard to debug in the future, so I'd advise refactoring it into several separate commands. 

```suggestion
            lambda x: any(glob.fnmatch.fnmatch(os.path.basename(x), match) for match in [DATASET_GLOB + SCAFFOLD_GLOB]),
            map(lambda x: x.filename, files)
        ))
```

Reply:
I don't think this is right since the variables are strings, this would combine the following two strings defined at the top of the file:
```
DATASET_GLOB = "dataset*.json*"
SCAFFOLD_GLOB = "scaffold*.json*"
```

Reply:
From the qca-dataset-submission management directory:
```
>>> any(glob.fnmatch.fnmatch(os.path.basename("validation.py"), match) for match in ["validation*", "lifecycle*"])
True
>>> any(glob.fnmatch.fnmatch(os.path.basename("lifecycle.py"), match) for match in ["validation*", "lifecycle*"])
True
>>> any(glob.fnmatch.fnmatch(os.path.basename("project*"), match) for match in ["validation*", "lifecycle*"])
False
```

Reply:
Ah, good catch. Not needed, then.